x-pack/solutions/observability/plugins/apm/public/components/app/dependencies_inventory/dependencies_inventory_table/index.tsx

```diff
@@ -27,6 +31,7 @@ export function DependenciesInventoryTable() {
   } = useApmParams('/dependencies/inventory');
   const { onPageReady } = usePerformanceContext();
   const { start, end } = useTimeRange({ rangeFrom, rangeTo });
+  const [renderedItems, setRenderedItems] = useState<FormattedSpanMetricGroup[]>([]);
```

Comment:
I think we need to reset `renderedItems` whenever a request is made to `GET /internal/apm/dependencies/top_dependencies`. This is because when clicking on refresh, the `/statistics` endpoint is called in parallel with `/top_dependencies`. 

<img width="690" alt="image" src="https://github.com/user-attachments/assets/df984576-3f5b-4ef7-9a15-35413e1e5cf7" />

Reply:
Yeah, but looks like, in this case, we are making an unnecessary request to `statistics` aren't we? I.e: it's fetching data for dependencies that will be replaced once `top_dependencies` responds. And once it responds, `statistics` will be called again for the actual dependencies it needs to fetch data for.

Reply:
>Yeah, but looks like, in this case, we are making an unnecessary request to statistics aren't we?

Not really, the stats API is only called once on every refresh. 
<img width="920" alt="Screenshot 2025-02-24 at 14 09 44" src="https://github.com/user-attachments/assets/bad54642-3f45-431b-81e9-ef25b34d3378" />

Reply:
🤔 . That's not what I'm seeing

https://github.com/user-attachments/assets/d74dc570-db90-4319-9aa8-6f7cd68bd2fe

Reply:
😲